### DISCUSSION TITLE
How to reveal content on paste reading page only?

### DISCUSSION THREAD

**Discussion by NKL-aatu** [PrivateBin/PrivateBin#1041] [Q&A]
I'd like to add some extra instructions to paste readers, by which I mean the people who've received the paste link. This would require to set that extra div element "hidden" by default and unhide it if the finished paste is visible (paste link is used).

Do you have any suggestions on what's the best way to accomplish this?

- Is there a php or js function that tells if we're on the composing or reading view that I can use to toggle between hide/unhide?
- Should I extend some class to include this extra hidden/unhide change?
- Or is there some specific setting on the page or in the cookie that's present only in the reading view and which I can use as the basis of my custom function (like: if A is "hidden", then B is unhidden)?

**Comment by elrido** [comment]
I looked into it and we do use different elements with specific IDs to hide and un-hide the different styled messages. You would need to add your div in your template and also change the javascript logic slightly, so that it gets enabled alongside the message.

The new element with class "hidden" and an ID to target it, would be placed in this area (above or below):
https://github.com/PrivateBin/PrivateBin/blob/e84a8694e4f4ca6ff7dd745aa985e27388d3dad0/tpl/bootstrap.php#L556-L566

And you'd need to add a statement to `$('#yourElementsID').removeClass('hidden');` in the `showPaste()` method:
https://github.com/PrivateBin/PrivateBin/blob/e84a8694e4f4ca6ff7dd745aa985e27388d3dad0/js/privatebin.js#L2598-L2625

Optionally, if you want to hide it again, when a new paste is created or the state otherwise changes, you could hook into matching `hide()` method:
https://github.com/PrivateBin/PrivateBin/blob/e84a8694e4f4ca6ff7dd745aa985e27388d3dad0/js/privatebin.js#L2720-L2738

Finally, if you do change the javascript file, you will need to [update it's new SRI hash in the template file](https://github.com/PrivateBin/PrivateBin/wiki/Development#subresource-integrity-for-javascript-resources).

**Reply by NKL-aatu** [reply]
Thanks. I'd prefer not to update PrivateBin's core, so that I wouldn't have to worry about updates. But I'll do it, if necessary. Thanks for giving me specific instructions.

Did I understand correctly that when the "reading view" is displayed there are NO elements that would be always hidden / unhidden in that view and the other way in all other views ("editing view" and "finished view"). For me it would be easier to add an event listener or st on that HTML element and use that to trigger the unhiding of the new div than adjusting the privatebin.js.

**Reply by elrido** [reply]
Using an even listener sounds indeed less intrusive. The IDs of the two elements is fixed, has not changed since back in the ZeroBin days and is the same across all templates. You'd need to monitor the two IDs "prettymessage" and "plaintext". If either of them gets displayed (= attribute class "hidden" removed) you'd attach your element, before or after it.

Which one is displayed depends on the format of the paste, source code ones use the "prettymessage", while plain text and markdown ones use the other.

**Reply by NKL-aatu** [reply]
Thank you, I wouldn't have thought of monitoring both of them! I'll try it.
